lambda/health_package/metric-settings.json

```diff
@@ -46,38 +248,410 @@
+  },
+  {
+    "Namespace": "AWS/Firehose",
+    "MetricName": "PutRecords.FailedRecords",
+    "Statistic": "Average",
+    "ComparisonOperator": "LessThanThreshold",
+    "EvaluationPeriods": 2,
+    "Period": 300,
+    "Multiplier": 1.1,
+    "Minimum": 0.99,
+    "Description": "The number of records rejected due to internal failures in a PutRecords operation per Kinesis data stream, measured over the specified time period. Occasional internal failures are to be expected and should be retried."
+  },
+  {
+    "Namespace": "AWS/Firehose",
+    "MetricName": "PutRecords.Latency",
+    "Statistic": "Average",
+    "ComparisonOperator": "GreaterThanOrEqualToThreshold",
+    "EvaluationPeriods": 2,
+    "Period": 300,
+    "Multiplier": 1.1,
+    "Minimum": 2,
+    "Maximum": 60,
+    "Description": "The time taken (msecs) per PutRecords operation, measured over the specified time period."
+  },
+  {
+    "Namespace": "AWS/Firehose",
+    "MetricName": "PutRecords.Success",
+    "Statistic": "Minimum",
+    "ComparisonOperator": "LessThanThreshold",
+    "EvaluationPeriods": 2,
+    "Period": 300,
+    "Multiplier": 1.1,
+    "Minimum": 0.99,
+    "Description": "The number of PutRecords operations where at least one record succeeded, per Kinesis stream, measured over the specified time period."
+  },
+  {
+    "Namespace": "AWS/Firehose",
+    "MetricName": "PutRecords.SuccessfulRecords",
+    "Statistic": "Minimum",
+    "ComparisonOperator": "LessThanThreshold",
+    "EvaluationPeriods": 2,
+    "Period": 300,
+    "Multiplier": 1.1,
+    "Minimum": 0.99,
+    "Description": "The number of successful records in a PutRecords operation per Kinesis data stream, measured over the specified time period."
+  },
+  {
+    "Namespace": "AWS/Firehose",
+    "MetricName": "PutRecords.ThrottledRecords",
+    "Statistic": "Maximum",
+    "ComparisonOperator": "GreaterThanOrEqualToThreshold",
+    "EvaluationPeriods": 2,
+    "Period": 300,
+    "Multiplier": 1,
+    "Minimum": 0.99,
+    "Description": "The number of records rejected due to throttling in a PutRecords operation per Kinesis data stream, measured over the specified time period."
+  },
+  {
+    "Namespace": "AWS/Firehose",
+    "MetricName": "ReadProvisionedThroughputExceeded",
+    "Statistic": "Minimum",
+    "ComparisonOperator": "GreaterThanOrEqualToThreshold",
+    "EvaluationPeriods": 2,
+    "Period": 300,
+    "Multiplier": 1,
+    "Minimum": 0.99,
+    "Description": "When the Minimum statistic has a value of 1, all records were throttled for the stream during the specified time period."
+  },
+  {
+    "Namespace": "AWS/Firehose",
+    "MetricName": "WriteProvisionedThroughputExceeded",
+    "Statistic": "Minimum",
+    "ComparisonOperator": "GreaterThanOrEqualToThreshold",
+    "EvaluationPeriods": 2,
+    "Period": 300,
+    "Multiplier": 1,
+    "Minimum": 0.99,
```

Comment:
> When the Minimum statistic has a non-zero value, records were being throttled for the stream during the specified time period.

The way I read that we should be aiming for this stat to be 0 and alerting if it's ever more than 0. I think this minimum says we won't alert until at least 99% of records are being throttled. 

I think for this we want is `EvaluationPeriods: 1`, `Minimum: 0`, `Maximum: 0`, `Statistic: Maximum`,  `ComparisonOperator: GreaterThanOrEqualToThreshold` - So always set the threshold to 0 and always alert if the maximum value in the evaluation period is non-zero. 

If you set it to minimum then it could be 0 for 5 seconds and 1 for 4m:55s and it would report 0 and not trigger the alarm.

Reply:
Thanks that part of the documentation confused the hell out of me.

Reply:
Me too. I'm not convinced I'm right. The docs are awful.